tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -600,24 +600,51 @@ private async Task<string[]> GetObjectsPackFileFromWorkingDirectoryAsync(string[
         var temporaryPath = Path.GetTempFileName();
 
         var getObjectsArguments = "rev-list --objects --no-object-names --filter=blob:none --since=\"1 month ago\" HEAD " + string.Join(" ", commitsToExclude.Select(c => "^" + c));
-        var getObjects = await ProcessHelpers.RunCommandAsync(new ProcessHelpers.Command("git", getObjectsArguments, _workingDirectory)).ConfigureAwait(false);
-        if (string.IsNullOrEmpty(getObjects))
+        var getObjectsCommand = await ProcessHelpers.RunCommandAsync(new ProcessHelpers.Command("git", getObjectsArguments, _workingDirectory)).ConfigureAwait(false);
+        if (string.IsNullOrEmpty(getObjectsCommand?.Output))
         {
             // If not objects has been returned we skip the pack + upload.
             Log.Debug("ITR: No objects were returned from the git rev-list command.");
             return Array.Empty<string>();
         }
 
         Log.Debug("ITR: Packing objects...");
-        var getPacksArguments = $"pack-objects --compression=9 --max-pack-size={MaxPackFileSizeInMb}m  {temporaryPath}";
-        var packObjectsResult = await ProcessHelpers.RunCommandAsync(new ProcessHelpers.Command("git", getPacksArguments, _workingDirectory), getObjects).ConfigureAwait(false);
-        if (packObjectsResult is null)
+        var getPacksArguments = $"pack-objects --compression=9 --max-pack-size={MaxPackFileSizeInMb}m \"{temporaryPath}\"";
+        var packObjectsResultCommand = await ProcessHelpers.RunCommandAsync(new ProcessHelpers.Command("git", getPacksArguments, _workingDirectory), getObjectsCommand!.Output).ConfigureAwait(false);
+        if (packObjectsResultCommand is null)
         {
             Log.Warning("ITR: 'git pack-objects...' command is null");
             return Array.Empty<string>();
         }
 
-        var packObjectsSha = packObjectsResult.Split(new[] { "\n" }, StringSplitOptions.RemoveEmptyEntries);
+        if (packObjectsResultCommand.ExitCode != 0 && packObjectsResultCommand.Error.IndexOf("Cross-device", StringComparison.OrdinalIgnoreCase) != -1)
+        {
+            // Git can throw a cross device error if the temporal folder is in a different drive than the .git folder (eg. symbolic link)
+            // to handle this edge case, we create a temporal folder inside the current folder.
+
+            Log.Warning("ITR: 'git pack-objects...' returned a cross-device error, retrying using a local temporal folder.");
+            var temporaryFolder = Path.Combine(Environment.CurrentDirectory, ".git_tmp");
```

Comment:
Do you have to create the temporary folder in the current directory? Would it be better/safer to create it in the temp directory (or at least try that first). Also, should probably try to _delete_ that folder afterwards? Otherwise it could (for example) mess up any `git diff` or similar commands they're running in CI?

Reply:
well, is the first thing I try in the code, getting a temporal folder/file from the dotnet api, and this is the fallback, only runs when the temp folder doesn't work.

You're right about the `delete` thing, I'll do that.